### PR TITLE
Distribution in dev pool

### PR DESCRIPTION
**Pull request by abolinbsu**

### CODE REVIEW

File: libraries/app/include/scorum/app/scorum_api_objects.hpp [hunk 1]
```diff
@@ -499,7 +499,7 @@ struct budget_api_obj
     asset balance = asset(0, SCORUM_SYMBOL);
     share_type per_block;
 
-    uint32_t last_allocated_block;
+    uint32_t last_cashout_block;
```
Comment (alexplevako) [inline libraries/app/include/scorum/app/scorum_api_objects.hpp]: could we initialize it? ```uint32_t last_cashout_block = 0;```

File: libraries/chain/database/block_tasks/process_funds.cpp [hunk 2]
```diff
@@ -26,31 +28,43 @@ void process_funds::on_apply(block_task_context& ctx)
...
+    {
+        original_fund_reward += budget_service.allocate_cash(budget);
+    }
 
-    auto content_reward = asset(total_block_reward.amount * SCORUM_CONTENT_REWARD_PERCENT / SCORUM_100_PERCENT,
-                                total_block_reward.symbol());
-    auto witness_reward = total_block_reward - content_reward; /// Remaining 5% to witness pay
+    // 50% of the revenue goes to support and develop the product, namely,
+    // towards the company�s R&D center.
+    auto dev_team_reward = advertising_budgets_reward * SCORUM_DEV_TEAM_PER_BLOCK_REWARD_PERCENT / SCORUM_100_PERCENT;
```
Comment (alexplevako) [inline libraries/chain/database/block_tasks/process_funds.cpp]: I would like to explicit type name

File: tests/common/database_trx_integration.cpp [hunk 3]
```diff
@@ -22,6 +22,8 @@
 
 #include "database_trx_integration.hpp"
 
+#define SCORUM_NUM_INIT_DELEGATES 1
```
Comment (Andrei-Masilevich) [inline tests/common/database_trx_integration.cpp]: Move to defines.hpp near:

#define TEST_INIT_DELEGATE_NAME "initdelegate"
#define TEST_INIT_KEY TEST_INIT_DELEGATE_NAME

File: libraries/protocol/include/scorum/protocol/config.hpp [hunk 4]
```diff
@@ -141,8 +139,9 @@
 #define SCORUM_1_TENTH_PERCENT                 (SCORUM_100_PERCENT/1000)
 #define SCORUM_PERCENT(X)                      (uint16_t)(X*SCORUM_1_PERCENT)
 
-#define SCORUM_CONTENT_REWARD_PERCENT          (95*SCORUM_1_PERCENT)
-#define SCORUM_CURATION_REWARD_PERCENT         (25*SCORUM_1_PERCENT)
+#define SCORUM_DEV_TEAM_PER_BLOCK_REWARD_PERCENT    SCORUM_PERCENT(50)
```
Comment (alexplevako) [inline libraries/protocol/include/scorum/protocol/config.hpp]: SCORUM_DEVELOPMENT_POOL_PER_BLOCK_REWARD_PERCENT

File: libraries/chain/services/budget.cpp [hunk 5]
```diff
@@ -106,75 +103,72 @@ const budget_object& dbs_budget::create_fund_budget(const asset& balance, const
...
+    dbs_dynamic_global_property& dgp_service = db().obtain_service<dbs_dynamic_global_property>();
+
+    const dynamic_global_property_object& dprops = dgp_service.get();
+
+    // clang-format off
+    FC_ASSERT(owner.name != SCORUM_ROOT_POST_PARENT_ACCOUNT, "'${1}' name is not allowed for ordinary budget.",("1", SCORUM_ROOT_POST_PARENT_ACCOUNT));
     FC_ASSERT(balance.symbol() == SCORUM_SYMBOL, "Invalid asset type (symbol).");
     FC_ASSERT(balance.amount > 0, "Invalid balance.");
     FC_ASSERT(owner.balance >= balance, "Insufficient funds.");
-    FC_ASSERT(_get_budget_count(owner.name) < SCORUM_BUDGET_LIMIT_COUNT_PER_OWNER,
```
[the remark this answers is not in the document]
Comment (abolinbsu) [reply libraries/chain/services/budget.cpp]: Your name supposes money limit, not quantity

File: libraries/protocol/include/scorum/protocol/config.hpp (hunk 4 above)
Comment (abolinbsu) [reply libraries/protocol/include/scorum/protocol/config.hpp]: Reward is aimed for dev team like(reward for witnesses and curators) not for dev pool

File: libraries/chain/services/budget.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (AlexShkor) [reply libraries/chain/services/budget.cpp]: better SCORUM_BUDGETS_LIMIT_PER_OWNER

File: libraries/chain/database/block_tasks/process_funds.cpp [hunk 6]
```diff
@@ -42,10 +42,14 @@ void process_funds::on_apply(block_task_context& ctx)
         advertising_budgets_reward += budget_service.allocate_cash(budget);
     }
 
-    for (const budget_object& budget : budget_service.get_fund_budgets())
+    try
     {
+        const budget_object& budget = budget_service.get_fund_budget();
         original_fund_reward += budget_service.allocate_cash(budget);
     }
+    catch (...)
```
Comment (Andrei-Masilevich) [inline libraries/chain/database/block_tasks/process_funds.cpp]: I think that is_exists would be better.

**Pull request merged**